app/controllers/admin/chat_channels_controller.rb

```diff
@@ -29,6 +29,17 @@ def remove_user
       redirect_back(fallback_location: "/admin/chat_channels")
     end
 
+    def destroy
+      @chat_channel = ChatChannel.find(params[:id])
+      if @chat_channel.users.count.zero?
```

Comment:
small optimization: `.count` is going to issue a SQL query no matter how many times it's called in a request.
As this is called both in the controller and in the view, I think we can save one SQL query by caching this:

```
@chat_channel_users_count = @chat_channel.users.count
if @chat_channel_users_count.zero?
```

and then in the view you can use the same variable

Reply:
nvm, this would apply to a potential `index` action which counts both in the controller and in the view. There's no view here :D